packages/shared/src/components/BlockItemWrapper/BlockItemWrapper.tsx

```diff
@@ -14,21 +14,24 @@ export const BlockItemWrapper = ({
     isDragging,
 }: PropsWithChildren<BlockItemWrapperProps>) => {
     const [isFlyoutOpen, setIsFlyoutOpen] = useState(false);
+
     return (
         <div
+            tabIndex={0}
             data-test-id="block-item-wrapper"
             className={joinClassNames([
-                'tw-relative tw-group',
-                !shouldHideWrapper &&
-                    'hover:tw-outline-offset-[2px] hover:tw-outline hover:tw-outline-[1px] hover:tw-outline-box-selected-inverse',
+                'tw-relative tw-group tw-outline-offset-2 tw-outline-1 tw-outline-box-selected-inverse',
+                !shouldHideWrapper && ' hover:tw-outline focus:tw-outline focus-within:tw-outline',
+                isFlyoutOpen && ' tw-outline',
             ])}
         >
             <div className={joinClassNames([shouldHideComponent && 'tw-opacity-0'])}>
                 <div
                     className={joinClassNames([
-                        'tw-absolute tw-bottom-[calc(100%-4px)] tw-right-[-3px] tw-w-full',
-                        !shouldHideWrapper && 'group-hover:tw-opacity-100',
-                        isFlyoutOpen && !shouldHideWrapper ? 'tw-opacity-100' : 'tw-opacity-0',
+                        'tw-absolute tw-bottom-[calc(100%-4px)] tw-right-[-3px] tw-w-full tw-opacity-0',
+                        !shouldHideWrapper &&
+                            ' group-hover:tw-opacity-100 group-focus:tw-opacity-100 focus-within:tw-opacity-100',
+                        isFlyoutOpen && ' tw-opacity-100',
```

Comment:
Nit: I don't think there is need for a space at the beginning:
```suggestion
                            'group-hover:tw-opacity-100 group-focus:tw-opacity-100 focus-within:tw-opacity-100',
                        isFlyoutOpen && 'tw-opacity-100',
```